spyderlib/widgets/dicteditor.py

```diff
@@ -1075,7 +1076,23 @@ def copy(self):
         for idx in self.selectedIndexes():
             if not idx.isValid():
                 continue
-            clipl.append(to_text_string(self.delegate.get_value(idx)))
+            obj = self.delegate.get_value(idx)
+            # check if it's a numpy array, and if so make sure to copy the whole
+            # thing in a tab separated format
+            try:
+                import numpy as np
+            except ImportError:
+                pass # skip this part
+            else:
+                if isinstance(obj, np.ndarray):
+                    if PY3:
+                        output = io.BytesIO()
+                    else:
+                        output = io.StringIO()
+                    np.savetxt(output, obj, delimiter='\t')
+                obj = output.getvalue().decode('utf-8')
+                    
+            clipl.append(to_text_string(obj))

```

Comment:
This is use of `to_text_string` is what is not needed, i.e. this line needs to be

    clipl.append(obj)

because `obj` is already in `utf-8`